pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -135,36 +121,32 @@ func (nt *nodeTree) updateNode(old, new *v1.Node) {
 	nt.addNode(new)
 }
 
-func (nt *nodeTree) resetExhausted() {
-	for _, na := range nt.tree {
-		na.lastIndex = 0
-	}
-	nt.zoneIndex = 0
-}
-
-// next returns the name of the next node. NodeTree iterates over zones and in each zone iterates
+// listNodes returns the list of namesof the node. NodeTree iterates over zones and in each zone iterates
 // over nodes in a round robin fashion.
-func (nt *nodeTree) next() string {
+func (nt *nodeTree) list() ([]string, error) {
+	nodesList := make([]string, 0, nt.numNodes)
 	if len(nt.zones) == 0 {
```

Comment:
nit: `return nil, nil`, before making the allocation in the line before.

Reply:
should it return nil or an empty list ?

Reply:
nil is equivalent to empty list

Reply:
fixed, thanks